apps/authentication/api/views.py

```diff
@@ -83,6 +85,44 @@ def dump_data(self, request, pk: int):
         serializer = self.get_serializer(user)
         return Response(data=serializer.data, status=status.HTTP_200_OK)
 
+    @action(detail=True, methods=["get"], url_path="permissions")
+    def user_permissions(self, request, pk):
+        """
+        Returns all permissions for a given user.
+        Codename is "action_object", e.g. "delete_profile".
+        Content_type is ID of object type.
+        If you want to group permissions, you can group on object type and map the ID to the object from the codename string.
+        """
+        user = self.get_object()
+        permissions = (
+            Permission.objects.filter(group__user=user)
+            .order_by("content_type")
+            .distinct()
```

Comment:
Won't this just get the permissions assigned to a user through a group?
A user may have permissions assigned to themselves as well?

Reply:
You're right, trying to find out how I can find all permissions by user while still using the Permission-object.

Reply:
I think there is a `get_all_permissions`, which also returns strings.

Reply:
Resolved in f4d3847